packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,242 @@
+
+    console.log(bold(cyanBright(`ⓘ  ${msg}`)));
+    process.stdout.write('\n');
+
+    readline.cursorTo(process.stdout, 0, totalRows - 2);
+
+    // for current status
+    if (status) {
+        process.stdout.write(`${green('⬤')}  `);
+    } else {
+        process.stdout.write(`${red('⬤')}  `);
+    }
+};
+
+/**
+ * Clear the whole terminal
+ */
+const clrscr = () => {
+    process.stdout.write('\x1B[2J\x1B[3J\x1B[H');
+};
+
+/**
+ * Helper plugin for child compilers if MultiCompiler is supplied
+ */
+class InteractiveModeMultiCompilerHelperPlugin {
+    constructor() {
+        this.name = 'InteractiveModeMultiCompilerHelperPlugin';
+    }
+
+    apply(compiler) {
+        // clear terminal if any one of child starts compilation
+        compiler.hooks.beforeCompile.tap(this.name, () => {
+            clrscr();
+            compiler.hooks.beforeInteractiveStats.call();
+        });
+    }
+}
+
+const isWebpack5 = version.startsWith('5');
+
+/**
+ * Interactive Mode plugin
+ */
+class InteractiveModePlugin {
+    constructor() {
+        this.isMultiCompiler = false;
+        this.compilers = undefined;
+        this.name = 'InteractiveModePlugin';
+        this.watching = undefined;
+        this.keys = {
+            quit: 'q',
+            stop: 's',
+            start: 'w',
+        };
+        this.handlers = {
+            quit: this.quitHandler.bind(this),
+            stop: this.stopHandler.bind(this),
+            start: this.startHandler.bind(this),
+        };
+    }
+
+    apply(compiler) {
+        if (!isWebpack5) {
+            logger.error('Interactive is not supported on webpack v4 and less');
+            process.exit(1);
+        }
+
+        // Configure stdin for keypress event
+        const stdin = process.stdin;
+        stdin.setEncoding('utf-8');
+        stdin.setRawMode(true);
+        readline.emitKeypressEvents(stdin);
+
+        // Configure keypress event for actions
+        const actions = Object.keys(this.keys);
+        stdin.on('keypress', (_, actionKey) => {
+            const possibleActions = actions.filter((action) => {
+                return this.keys[action] === actionKey.name;
+            });
+
+            if (possibleActions.length === 0) {
+                return;
+            }
+
+            if (possibleActions.length > 1) {
+                throw new Error('Multiple actions are provided for same key');
+            }
+
+            const action = possibleActions[0];
+            this.handlers[action](compiler);
+        });
+
+        if (compiler.compilers) {
+            this.isMultiCompiler = true;
+            this.compilers = compiler.compilers;
+        }
+
+        // Register Custom Hook for printing after clrscr
+        if (!compiler.hooks.beforeInteractiveStats) {
+            compiler.hooks = {
+                ...compiler.hooks,
+                beforeInteractiveStats: new SyncHook(),
+            };
+        }
+
+        // Clear for first run as well
+        clrscr();
```

Comment:
maybe bad idea, we can remove extra logs here (from external tools), better clean screen only on `beforeCompile` hooks